src/credentials-manager/index.js

```diff
@@ -0,0 +1,145 @@
+   * @param {String} credentials.refreshToken optional - used to refresh access token
+   * @param {String} credentials.scope optional - represents the scope of the current token
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async saveCredentials(credentials = {}) {
+    const validateKeys = ['idToken', 'accessToken', 'tokenType'];
+    validateKeys.forEach(key => {
+      if (!credentials[key]) {
+        const json = {
+          error: 'a0.credential_manager.invalid_input',
+          error_description: `${key} cannot be empty`,
+          invalid_parameter: key,
+        };
+        throw new CredentialsManagerError({json, status: 0});
+      }
+    });
+    if (
+      (!credentials.expiresIn || !credentials.expiresIn > 0) &&
+      !credentials.expiresAt
+    ) {
+      const json = {
+        error: 'a0.credential_manager.invalid_input',
+        error_description: `expiresIn or expiresAt should be set`,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+    try {
+      await this.ensureCredentialManagerIsInitialized();
+      await this.Auth0Module.saveCredentials(credentials);
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credential that has already been saved
+   *
+   * @param {String} scope optional - the scope to request for the access token. If null is passed, the previous scope will be kept.
+   * @param {String} minTtl optional - the minimum time in seconds that the access token should last before expiration.
+   * @param {Object} parameters optional - additional parameters to send in the request to refresh expired credentials.
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async getCredentials(scope, minTtl = 0, parameters = {}) {
+    try {
+      await this.ensureCredentialManagerIsInitialized();
+      const credentials = await this.Auth0Module.getCredentials(
+        scope,
+        minTtl,
+        parameters,
+      );
+      return credentials;
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credential that has already been saved
+   *
+   * @param {String} title optional - the text to use as title in the authentication screen. Passing null will result in using the OS's default value.
+   * @param {String} description optional - the text to use as description in the authentication screen. On some Android versions it might not be shown. Passing null will result in using the OS's default value.
```

Comment:
> On some Android versions it might not be shown

Why not? Should we link to more information?

Reply:
This would be because of how the device vendor has designed this particular screen Steve. I don't think that needs to be called out here